### PR TITLE
New structure, refactoring, name change, makefile, README update

### PR DESCRIPTION
**Pull request by theBadMusician**
- Created appropriate C code structure.
- Split separate functions/functionalities into respective header and source files.
- Using name ATMEGA2560_THR_INF_Firmware thoughout the repo.
- Added a makefile to compile the codebase using AVR-GCC and flash the MCU using AVRDUDE commands. (No Arduino IDE required)
- Updated README.md. Note: should be expanded with more detailed instructions in the future.

### CODE REVIEW

File: src/Thrusters.c [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+		break;
+	}
+}
+
+void set_thrust_all(uint8_t thrust)
+{
+    for (uint8_t i = 0; i < NUM_THRUSTERS; i++) set_thrust(i, thrust);
+}
+
+void prearm_thruster_flood()
```
Comment (oyssolbo) [inline src/Thrusters.c:119]: What does "..._flood" stand for?

**Review by oyssolbo — CHANGES_REQUESTED** [review]
Impressed by the makefile! Very good! 

Mostly nitpicking on comments. You will hate me for this, but we should write strict doxygen when documenting C/C++. This makes it easier later when someone external looks through the code. Also, the pragma once should be discarded since we already use ifndef/define, and pragma once is not well-defined for C, only C++

**Comment by oyssolbo** [comment]
And also, I'm a bit biased since I participated in writing this code

**Comment by theBadMusician** [comment]
Fixed problems mentioned in the review.

**Pull request merged**